src/content/app/tools/blast/components/blast-input-sequences/useBlastInputSequences.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useRef, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import {
+  setSequences,
+  setSequenceType
+} from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+
+import {
+  getSequences,
+  getSelectedSequenceType,
+  getSequenceSelectionMode
+} from 'src/content/app/tools/blast/state/blast-form/blastFormSelectors';
+
+import { guessSequenceType } from 'src/content/app/tools/shared/helpers/sequenceTypeGuesser';
+
+import untypedBlastSettingsConfig from 'src/content/app/tools/blast/components/blast-settings/blastSettingsConfig.json';
+
+import type {
+  SequenceType,
+  BlastSettingsConfig
+} from 'src/content/app/tools/blast/types/blastSettings';
+import type { ParsedInputSequence } from 'src/content/app/tools/blast/types/parsedInputSequence';
+
+const blastSettingsConfig = untypedBlastSettingsConfig as BlastSettingsConfig;
+
+const useBlastInputSequences = () => {
+  const sequences = useSelector(getSequences);
+  const sequenceType = useSelector(getSelectedSequenceType);
+  const sequenceSelectionMode = useSelector(getSequenceSelectionMode);
+  const dispatch = useDispatch();
+
+  const ref = useRef({
+    sequenceType,
+    sequenceSelectionMode
+  });
+
+  // to avoid stale closures persisting in functions returned from the hook
+  useEffect(() => {
+    ref.current = { sequenceType, sequenceSelectionMode };
+  });
+
+  const updateSequences = (sequences: ParsedInputSequence[]) => {
+    dispatch(setSequences({ sequences }));
+    updateSequenceTypeAutomatically(sequences);
```

Comment:
Would it be better to avoid calling this function if `ref.current.sequenceSelectionMode === 'manual'` here instead of doing the check on L67?

Reply:
I don't think so. We also automatically update the sequence type to default when there are no sequences, sadly, regardless of whether the `sequenceSelectionMode` is automatic or manual. I think this check belongs in the function that is concerned with the updating of sequence type rather than with the updating of the sequences.